org.jacoco.core/src/org/jacoco/core/internal/BytecodeVersion.java

```diff
@@ -0,0 +1,78 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2018 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.internal;
+
+import org.objectweb.asm.Opcodes;
+
+/**
+ * Utilities to read and modify bytecode version in bytes of class.
+ */
+public final class BytecodeVersion {
+
+	private static final int VERSION_INDEX = 6;
+
+	/**
+	 * Version of the Java 10 class file format.
+	 */
+	public static final int V10 = Opcodes.V9 + 1;
+
+	private BytecodeVersion() {
+	}
+
+	/**
+	 * Reads bytecode version from given bytes of class.
+	 * 
+	 * @param b
+	 *            bytes of class
+	 * @return version of bytecode
+	 */
+	public static int read(final byte[] b) {
+		return (short) (((b[VERSION_INDEX] & 0xFF) << 8)
+				| (b[VERSION_INDEX + 1] & 0xFF));
+	}
+
+	/**
+	 * Sets bytecode version in given bytes of class.
+	 *
+	 * @param b
+	 *            bytes of class
+	 * @param version
+	 *            version of bytecode to set
+	 */
+	public static void set(final byte[] b, final int version) {
```

Comment:
For symmetry reasons I would prefer read/write (or get/set) for the method names.

Reply:
@marchof renamed `read` into `get`